src/iexamine.cpp

```diff
@@ -6514,41 +6601,57 @@ static void mill_activate( Character &you, const tripoint_bub_ms &examp )
 }
 
 static void smoker_activate( Character &you, const tripoint_bub_ms &examp )
+{
+    if( !iexamine::smoker_prep( you, examp ) ) {
+        return;
+    };
+
+    pick_firestarter_and_fire( you, examp, firestarter_actor::start_type::SMOKER );
+}
+
+static void smoker_prep_internal(
+    const tripoint_bub_ms &examp,
+    item **charcoal_ref,
+    std::reference_wrapper<units::volume> food_volume_ref
+)
```

Comment:
I should have thought about it the last round of review, but we can also make this signature much less scary as well by returning an `std::pair<item *, units::volume>`